File: 3474.lexicographically-smallest-generated-string.py
#
# @lc app=leetcode id=3474 lang=python3
#
# [3474] Lexicographically Smallest Generated String
#

# @lc code=start
class Solution:
    def generateString(self, str1: str, str2: str) -> str:
        n = len(str1)
        m = len(str2)
        result = [''] * (n + m - 1)
        i = 0
        while i <= n - m:
            if str1[i] == 'T':
                result[i:i+m] = list(str2)
            else:
                # Construct a non-matching segment by altering a character in str2 minimally.
                non_matching_segment = list(str2)
                # Try replacements using multiple alternatives to ensure non-match and minimal lexicographical impact.
                for j in range(m):
                    for char in 'abcdefghijklmnopqrstuvwxyz':
                        if char != str2[j]:
                            non_matching_segment[j] = char
                            break
                    break  # Only change one character to ensure minimal change.
                result[i:i+m] = non_matching_segment
            i += 1
        # Fill remaining parts with smallest lexicographical values if needed.
        for j in range(n+m-1):
            if result[j] == '':
                result[j] = 'a'
        # Verify if constructed result meets all conditions of T and F before returning.
        for k in range(n):
            current_substring = ''.join(result[k:k+m])
            if (str1[k] == 'T' and current_substring != str2) or (str1[k] == 'F' and current_substring == str2):
                return ""
        return ''.join(result)
# @lc code=end

File: 3474.lexicographically-smallest-generated-string.go
func generateString(str1 string, str2 string) string {
    n, m := len(str1), len(str2)
    word := make([]byte, n+m-1)
    for i := range word {
        word[i] = 'a' // Use 'a' as default
    }

    // Process each character in str1
    for i := 0; i < n; i++ {
        if str1[i] == 'T' {
            // Ensure substring matches str2 starting at i
            for j := 0; j < m && (i+j) < len(word); j++ {
                word[i+j] = str2[j]
            }
        } else { // str1[i] == 'F'
            // Ensure no matching substring starts at i
            mismatch := false
            for j := 0; j < m && (i+j) < len(word); j++ {
                if word[i+j] == str2[j] {
                    mismatch = true
                    break
                }
            }
            if !mismatch {
                for j := 0; j < m && (i+j) < len(word); j++ {
                    if word[i+j] == str2[j] {
                        // Change to a non-matching character
                        word[i+j] = byte((int(word[i+j]-'a')+1)%26 + 'a')
                    }
                }
            }
        }
    }
    return string(word)
}